Il2CppInspector.Common/Outputs/Namespace.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+    Copyright 2017-2020 Katy Coe - http://www.hearthcode.org - http://www.djkaty.com
+    Copyright 2020 Robert Xiao - https://robertxiao.ca
+
+    All rights reserved.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Il2CppInspector.Outputs
+{
+    /// <summary>
+    /// A utility class for managing names in a common namespace.
+    /// </summary>
+    public class Namespace
+    {
+        private readonly Dictionary<string, int> renameCount = new Dictionary<string, int>();
+
+        public void ReserveName(string name) {
+            if (renameCount.ContainsKey(name)) {
+                throw new Exception($"Can't reserve {name}: already taken!");
+            }
+            renameCount[name] = 0;
+        }
+
+        public Namer<T> MakeNamer<T>(Namer<T>.KeyFunc keyFunc) {
+            return new Namer<T>(this, keyFunc);
+        }
+
+        /// <summary>
+        /// A class for managing objects of a common type within a namespace.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public class Namer<T>
```

Comment:
This can be used to rename objects of any type - such as strings, enum members, etc., which can be useful if those will be mapped into a larger C namespace (e.g. the namespace of global identifiers).

Reply:
Fair enough 🙂